pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -260,24 +305,96 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 	if shoot.Status.LastMaintenance != nil && shoot.Status.LastMaintenance.State == gardencorev1beta1.LastOperationStateProcessing {
 		patch := client.MergeFrom(shoot.DeepCopy())
 		shoot.Status.LastMaintenance.State = gardencorev1beta1.LastOperationStateSucceeded
+
 		if err := r.Client.Status().Patch(ctx, shoot, patch); err != nil {
 			return err
 		}
 	}
 
-	for _, reason := range reasonForImageUpdatePerPool {
-		r.Recorder.Eventf(shoot, corev1.EventTypeNormal, gardencorev1beta1.ShootEventImageVersionMaintenance, "%s",
-			fmt.Sprintf("Updated %s.", reason))
+	// make sure to report (partial) maintenance failures
+	if reasonControlPlaneKubernetesUpdate != nil {
+		if successMsgControlPlaneKubernetesUpdate != nil {
+			r.Recorder.Eventf(shoot, corev1.EventTypeNormal, gardencorev1beta1.ShootEventK8sVersionMaintenance, "%s", fmt.Sprintf("Control Plane: %s. Reason: %s.",
+				*successMsgControlPlaneKubernetesUpdate, *reasonControlPlaneKubernetesUpdate))
```

Comment:
Inline it same as line 320.
```suggestion
			r.Recorder.Eventf(shoot, corev1.EventTypeNormal, gardencorev1beta1.ShootEventK8sVersionMaintenance, "%s", fmt.Sprintf("Control Plane: %s. Reason: %s.",*successMsgControlPlaneKubernetesUpdate, *reasonControlPlaneKubernetesUpdate))
```